test/unit/specs/events/cd-event-add-child-tickets.spec.js

```diff
@@ -0,0 +1,311 @@
+              invites: null,
+              name: 'Ticket2',
+              quantity: 1,
+              sessionId: 'sessionId',
+              totalApplications: 0,
+              type: 'ninja' }],
+        }];
+
+        // ACT
+
+        // ASSERT
+        expect(vm.childTickets).to.deep.equal([{ description: 'description1',
+          entity$: 'sessionentity',
+          eventId: 'some-eventId',
+          id: 'sessionId',
+          name: 'Session1',
+          status: 'status1',
+          tickets: [
+            { approvedApplications: 0,
+              deleted: 0,
+              entity$: 'ticketentity',
+              id: 'ticketId2',
+              invites: null,
+              name: 'Ticket2',
+              quantity: 1,
+              sessionId: 'sessionId',
+              totalApplications: 0,
+              type: 'ninja' }],
+        }]);
+      });
+    });
+
+    describe('computed.name', () => {
+      it('should return the full name - firstName and surname contactenated', async () => {
+        // ARRANGE
+        const vm = vueUnitHelper(ChildTicketWithMocks);
+        vm.firstName = 'Jane';
+        vm.surname = 'Doe';
+
+        // ACT
+
+        // ASSERT
+        expect(vm.name).to.equal('Jane Doe');
+      });
+    });
+
+    describe('computed.specifyGender', () => {
+      it('should return true if the selected gender has the value specify', async () => {
+        // ARRANGE
+        const vm = vueUnitHelper(ChildTicketWithMocks);
+        vm.genderSelect = 'specify';
+
+        // ACT
+
+        // ASSERT
+        expect(vm.specifyGender).to.equal(true);
+      });
+
+      it('should return false if the selected gender does not hav the value specify', async () => {
+        // ARRANGE
+        const vm = vueUnitHelper(ChildTicketWithMocks);
+        vm.genderSelect = 'male';
+
+        // ACT
+
+        // ASSERT
+        expect(vm.specifyGender).to.equal(false);
+      });
+    });
+
+    describe('computed.gender', () => {
+      it('should return the input gender value if the if the selected gender has the value specify', async () => {
+        // ARRANGE
+        const vm = vueUnitHelper(ChildTicketWithMocks);
+        vm.genderSelect = 'specify';
+        vm.genderInput = 'non-binary';
+
+        // ACT
+
+        // ASSERT
+        expect(vm.gender).to.equal('non-binary');
+      });
+
+      it('should return the selected gender value if the selected gender does not have the value specify', async () => {
+        // ARRANGE
+        const vm = vueUnitHelper(ChildTicketWithMocks);
+        vm.genderSelect = 'male';
+
+        // ACT
+
+        // ASSERT
+        expect(vm.gender).to.equal('male');
+      });
+    });
+
+    describe('computed.invalidTicket', () => {
+      it('should return the true if the ticketTouch is true and if the ticket length is equal to 0', async () => {
+        // ARRANGE
+        const vm = vueUnitHelper(ChildTicketWithMocks);
+        vm.ticketTouch = true;
+        vm.tickets = '';
```

Comment:
This works, but shouldn't it be an array rather than a string ?

Reply:
It should, that's my bad. Will fix that